docs/reference/search/suggesters/context-suggest.asciidoc

```diff
@@ -1,237 +1,269 @@
-type also allows to specify a field to extract the categories from. The `path` parameter is used to
-specify this field of the documents that should be used. If the referenced field contains multiple
-values, all these values will be used as alternative categories.
+
+The `category` context allows you to associate one or more categories to suggestions at index time.
+At query time, suggestions can be filtered and boosted by their associated categories.
 
 [float]
 ===== Category Mapping
 
-The mapping for a category is simply defined by its `default` values. These can either be
-defined as list of *default* categories:
+A `category` context mapping, where contexts are provided explicitly with suggestions
+can be defined as follows:
 
 [source,js]
 --------------------------------------------------
-"context": {
-    "color": {
+"contexts": [
+    {
+        "name": "cat_context",
         "type": "category",
-        "default": ["red", "orange"]
     }
-}
+]
 --------------------------------------------------
 
-or as a single value
+Alternatively, A `category` context mapping that references another field within a document
+can be defined as follows:
 
 [source,js]
 --------------------------------------------------
-"context": {
-    "color": {
+"contexts": [
+    {
+        "name": "cat_context",
         "type": "category",
-        "default": "red"
+        "path": "cat_field"
     }
-}
+]
 --------------------------------------------------
 
-or as reference to another field within the documents indexed:
+[float]
+===== Indexing category contexts
+
+`category` contexts can be explicitly set with suggestions or be indexed from another field in the
+document via the `path` parameter. Associating multiple contexts with a suggestion, will index the
+suggestion for every category. `category` contexts can be explicitly indexed with suggestions as follows:
 
 [source,js]
 --------------------------------------------------
-"context": {
-    "color": {
-        "type": "category",
-        "default": "red",
-        "path": "color_field"
+PUT place/shops/1
+{
+    ...
+    "suggest": {
+        "input": ["timmy's", "starbucks", "dunkin donuts"],
+        "context": {
+            "place_type": ["cafe", "food"]
+        }
     }
 }
 --------------------------------------------------
 
-in this case the *default* categories will only be used, if the given field does not
-exist within the document. In the example above the categories are received from a
-field named `color_field`. If this field does not exist a category *red* is assumed for
-the context *color*.
+NOTE: If context mapping references another field and the contexts
+are explicitly indexed, the suggestions are indexed with both set
+of contexts.
+
+`category` contexts can also be referenced from another indexed field in the document. The following
+will index the values of the 'cat' field, if the context mapping was configured with the `path`
```

Comment:
I'd provide an example of the mapping followed by the indexing request. Possibly also specify an explicit category, and add a callout `<1>` to say "This suggestion will be indexed with the following categories: [ explicit, cafe, food ]